.github/scripts/findUnusedStyles.ts

```diff
@@ -0,0 +1,462 @@
+import * as fs from 'fs';
+import {globSync} from 'glob';
+import * as path from 'path';
+import * as ts from 'typescript';
+
+type StyleDefinition = {
+    key: string;
+    file: string;
+    line: number;
+    column: number;
+};
+
+// Static patterns and constants - created once
+const STYLE_FILE_EXTENSIONS = '**/*.{ts,tsx,js,jsx}';
+const EXCLUDED_STYLE_DIRECTORIES = ['src/styles/utils/', 'src/styles/generators/', 'src/styles/theme/'];
+const VALID_IDENTIFIER_PATTERN = /^[a-zA-Z_$][a-zA-Z0-9_$]*$/;
+const SPREAD_PROPERTY_PATTERN = /\.\.\.\s*([a-zA-Z_$][a-zA-Z0-9_$]*)\s*\.\s*([a-zA-Z_$][a-zA-Z0-9_$]*)/g;
+const SIMPLE_SPREAD_PATTERN = /\.\.\.\s*([a-zA-Z_$][a-zA-Z0-9_$]*)\b(?!\s*\.)/g;
+const STYLE_KEY_SKIP_PATTERNS = ['default', 'exports', 'module', 'require', 'import', 'from', 'theme', 'colors', 'variables', 'CONST', 'Platform', 'StyleSheet'];
+
+// Comment removal patterns
+const SINGLE_LINE_COMMENT_PATTERN = /\/\/.*$/gm;
+const MULTI_LINE_COMMENT_PATTERN = /\/\*[\s\S]*?\*\//g;
+
+class ComprehensiveStylesFinder {
+    private rootDir: string;
+
+    private styleDefinitions = new Map<string, StyleDefinition>();
+
+    private fileContents = new Map<string, string>();
+
+    constructor(rootDir: string) {
+        this.rootDir = rootDir;
+    }
+
+    findUnusedStyles(): StyleDefinition[] {
+        console.log('🔍 Step 1: Finding all style definitions...');
+        this.findAllStyleDefinitions();
+        console.log(`📊 Found ${this.styleDefinitions.size} style definitions`);
+
+        console.log('🔍 Step 2: Loading all file contents...');
+        this.loadAllFileContents();
+
+        console.log('🔍 Step 3: Comprehensive style usage analysis...');
+        this.findStyleUsagesComprehensive();
+
+        console.log('📊 Step 4: Identifying unused styles...');
+        return this.getUnusedStyles();
+    }
+
+    private findAllStyleDefinitions(): void {
+        const styleFilesPaths = `src/styles/${STYLE_FILE_EXTENSIONS}`;
+        const styleFiles = globSync(styleFilesPaths, {
+            cwd: this.rootDir,
+        });
+
+        // Filter out utils, generators, and themes - focus only on main style files
+        const mainStyleFiles = styleFiles.filter((file) => {
```

Comment:
I think you can pass `ignore` as a param in the 2nd arg to `globSync` to simplify this.